src/core/src/pass/manager.cpp

```diff
@@ -89,89 +130,184 @@ class stopwatch {
-                               " requires static shape but the "
-                               "model is dynamic. Skipping this transformation");
-                continue;
+            bool is_pass_manager = name == m_manager_name;
+            if (is_pass_manager) {
+                std::cout << std::setw(25) << left;
+                std::cout << "PassManager started: " << m_manager_name << std::endl;
             }
+        }
+    }
+
+    void stop_timer(const std::string& name, bool applied) {
+        if (m_profile_pass.is_enabled()) {
+            auto& stopwatch = stopwatches.at(name);
+            stopwatch.stop();
 
-            if (dynamic_pointer_cast<Validate>(pass)) {
-                if (needs_validate) {
-                    function_pass->run_on_model(func);
-                    needs_validate = false;
+            bool is_pass_manager = name == m_manager_name;
+            if (m_profile_pass.is_bool()) {
+                std::cout << std::setw(25) << left;
+                if (is_pass_manager) {
+                    std::cout << "PassManager finished: ";
+                } else {
+                    std::cout << "  ";
+                }
+                std::cout << std::setw(60) << left << name;
+                std::cout << std::setw(5) << right << stopwatch.get_milliseconds() << "ms " << (applied ? "+" : "-")
+                          << std::endl;
+            } else if (m_file.is_open()) {
+                if (is_pass_manager) {
+                    m_file << "m;" << name << ";" << stopwatch.get_timer_value().count() << ";" << (applied ? "1" : "0")
+                           << std::endl;
+                } else {
+                    m_file << "t;" << name << ";" << m_manager_name << ";" << stopwatch.get_timer_value().count() << ";"
+                           << (applied ? "1" : "0") << std::endl;
                 }
             } else {
-                pass_applied = function_pass->run_on_model(func);
+                OPENVINO_THROW("The output file for logging transformation statistics is closed. "
+                               "Recording of statistics is not possible.");
             }
         }
+    }
 
-        if (m_visualize) {
-            // visualizations and serializations will be named after the outermost function
-            const size_t num_digits_in_pass_index = 3;
-            std::string index_str = std::to_string(index);
-            index_str = std::string(num_digits_in_pass_index - index_str.length(), '0') + index_str;
-            auto base_filename = func->get_name() + std::string("_") + index_str + std::string("_") + pass->get_name();
-
-            if (m_visualize) {
-                auto file_ext = "svg";
-                pass::VisualizeTree vt(base_filename + std::string(".") + file_ext);
-                vt.run_on_model(func);
+    void visualize(const shared_ptr<ov::Model>& model, const std::string& pass_name) const {
+        static size_t viz_index = 0;
```

Comment:
Make it private class member?

Reply:
I thought about it, but unfortunately in c++ 11/14 we can initialize private static members of class like this
`static size_t viz_index = 0;`
we have to split declaration and definition.

as these vars are used only inside the specific methods, I decided to leave a declaration and definition in one place.